### PR TITLE
Fix memory leaks on module remove

### PR DESCRIPTION
**Pull request by plbossart**
Tests with kmemleak show we don't release all resources/memory. Fix a couple, still quite a few to go.

### CODE REVIEW

File: sound/soc/codecs/hdac_hda.c [hunk 1]
```diff
@@ -518,6 +518,8 @@ static void hdac_hda_codec_remove(struct snd_soc_component *component)
 
 	pm_runtime_disable(&hdev->dev);
 	snd_hdac_ext_bus_link_put(hdev->bus, hlink);
+
+	snd_hdac_regmap_exit(hdev);
```
Comment (lyakh) [inline sound/soc/codecs/hdac_hda.c:524]: There's already a call to `snd_hdac_regmap_exit()` in `snd_hda_codec_cleanup_for_unbind()`, how do they play together?

Comment (plbossart) [reply sound/soc/codecs/hdac_hda.c:524]: we currently don't call that function so the regmap_exit() is never called.
The entire flow on remove() is not functional for HDaudio, Maybe this patch should be replaced by a better one, the point is that we don't release what we allocated.

Comment (kv2019i) [reply sound/soc/codecs/hdac_hda.c:524]: Hmm, the regmap_exit() is also missing in error goto label of hdac_hda_codec_probe(). 

@plbossart What do you mean by remove() being non-functional? I do test this and I can successfully unload hdac_hda.

Comment (plbossart) [reply sound/soc/codecs/hdac_hda.c:524]: yes you can unload but see what happens in terms of memory leaks: https://github.com/thesofproject/linux/issues/2195

Comment (kv2019i) [reply sound/soc/codecs/hdac_hda.c:524]: @plbossart ack, just double-checking there is no other brokenness on top. but yeah, the memleaks need to be fixed as well

Comment (plbossart) [reply sound/soc/codecs/hdac_hda.c:524]: fixed error handling in hdac_hda_codec_probe(), good suggestion @kv2019i

**Comment by plbossart** [comment]
Updated code with one more patch for a memory leak in regmap.

**Pull request merged**